build.gradle

```diff
@@ -163,41 +163,41 @@ tasks.create('testNaming') {
 	}
 }
 
-// Create a test task with given name, environments dir/file and is it development mode task
-void createTestTask(String name, String environments, boolean devMode) {
-	tasks.create(name) {
-		// Compile Skript and check test naming
+// Create a test task with given name, environments dir/file, dev mode and java version.
+void createTestTask(String name, String environments, boolean devMode, int javaVersion) {
+	tasks.register(name, JavaExec) {
 		dependsOn jar, testNaming
-		doFirst {
-			if (devMode) {
-				standardInput = System.in
-			}
-			javaexec {
-				group = 'execution'
-				classpath = files([
-					'build' + File.separator + 'libs' + File.separator + 'Skript.jar',
-					project.configurations.runtimeClasspath.find { it.name.startsWith('gson') },
-					sourceSets.main.runtimeClasspath
-				])
-				main = 'ch.njol.skript.tests.platform.PlatformMain'
-				args = [
-					'test_runners',
-					'src/test/skript/tests',
-					'src/test/resources/runner_data',
-					environments,
-					devMode
-				]
-			}
+		javaLauncher = javaToolchains.launcherFor {
+	        languageVersion = JavaLanguageVersion.of(javaVersion)
+	    }
+		if (devMode) {
+			standardInput = System.in
 		}
+		group = 'execution'
+		classpath = files([
+			'build' + File.separator + 'libs' + File.separator + 'Skript.jar',
+			project.configurations.runtimeClasspath.find { it.name.startsWith('gson') },
+			sourceSets.main.runtimeClasspath
+		])
+		main = 'ch.njol.skript.tests.platform.PlatformMain'
+		args = [
+			'test_runners',
+			'src/test/skript/tests',
+			'src/test/resources/runner_data',
+			environments,
+			devMode
+		]
 	}
 }
 
 // Register different Skript testing tasks
-createTestTask('quickTest', 'src/test/skript/environments/main/paper-1.18.2.json', false)
-createTestTask('skriptTest', 'src/test/skript/environments/main', false)
-createTestTask('skriptTestFull', 'src/test/skript/environments/', false)
+createTestTask('quickTest', 'src/test/skript/environments/main/paper-1.19.json', false, 17)
```

Comment:
There should be a constant for latest version instead of having to edit the build script IMO